src/main/java/com/synopsys/integration/alert/provider/blackduck/tasks/GroupSyncTask.java

```diff
@@ -0,0 +1,149 @@
+import com.synopsys.integration.blackduck.api.generated.view.UserGroupView;
+import com.synopsys.integration.blackduck.api.generated.view.UserView;
+import com.synopsys.integration.blackduck.service.HubService;
+import com.synopsys.integration.exception.IntegrationException;
+
+@Component
+public class GroupSyncTask extends SyncTask<GroupData> {
+    private final Logger logger = LoggerFactory.getLogger(GroupSyncTask.class);
+    private final BlackDuckUserRepositoryAccessor blackDuckUserRepositoryAccessor;
+    private final UserGroupRelationRepositoryAccessor userGroupRelationRepositoryAccessor;
+
+    @Autowired
+    public GroupSyncTask(final TaskScheduler taskScheduler, final BlackDuckProperties blackDuckProperties, final BlackDuckUserRepositoryAccessor blackDuckUserRepositoryAccessor,
+        final BlackDuckGroupRepositoryAccessor blackDuckGroupRepositoryAccessor,
+        final UserGroupRelationRepositoryAccessor userGroupRelationRepositoryAccessor) {
+        super(taskScheduler, "blackduck-sync-group-task", blackDuckProperties, blackDuckGroupRepositoryAccessor);
+        this.blackDuckUserRepositoryAccessor = blackDuckUserRepositoryAccessor;
+        this.userGroupRelationRepositoryAccessor = userGroupRelationRepositoryAccessor;
+    }
+
+    @Override
+    public List<UserGroupView> getHubViews(final HubService hubService) throws IntegrationException {
+        return hubService.getAllResponses(ApiDiscovery.USERGROUPS_LINK_RESPONSE);
+    }
+
+    @Override
+    public Map<GroupData, ? extends HubView> getCurrentData(final List<? extends HubView> hubViews) {
+        final List<UserGroupView> userGroupViews = (List<UserGroupView>) hubViews;
+        final Map<GroupData, ? extends HubView> groupMap = userGroupViews.stream().collect(Collectors.toMap(groupView -> new GroupData(groupView.name, groupView.active, groupView._meta.href), Function.identity()));
+        return groupMap;
+    }
+
+    @Override
+    public Set<GroupData> getStoredData(final List<? extends DatabaseEntity> storedEntities) {
+        final List<BlackDuckGroupEntity> blackDuckGroupEntities = (List<BlackDuckGroupEntity>) storedEntities;
+
+        final Set<GroupData> storedGroups = blackDuckGroupEntities.stream().map(blackDuckGroupEntity -> new GroupData(blackDuckGroupEntity.getName(), blackDuckGroupEntity.getActive(), blackDuckGroupEntity.getHref()))
+                                                .collect(Collectors.toSet());
+        return storedGroups;
+    }
+
+    @Override
+    public List<BlackDuckGroupEntity> getEntitiesToRemove(final List<? extends DatabaseEntity> storedEntities, final Set<GroupData> dataToRemove) {
+        final List<BlackDuckGroupEntity> blackDuckGroupEntities = (List<BlackDuckGroupEntity>) storedEntities;
+        final List<BlackDuckGroupEntity> blackDuckGroupsToRemove = blackDuckGroupEntities.stream()
+                                                                       .filter(blackDuckGroupEntity -> {
```

Comment:
When it comes to streams, do we want lambdas with more than one line (i.e. block-lambdas) to have their own method?